src/NuGet.Clients/PackageManagement.UI/PackageLoader.cs

```diff
@@ -283,6 +289,107 @@ private IEnumerable<string> GetSupportedFrameworks()
+                StringComparer.OrdinalIgnoreCase);
+
+            foreach (var project in _projects)
+            {
+                foreach (var package in (await project.GetInstalledPackagesAsync(token)))
+                {
+                    HashSet<NuGetVersion> versions;
+                    if (!packages.TryGetValue(package.PackageIdentity.Id, out versions))
+                    {
+                        versions = new HashSet<NuGetVersion>();
+                        packages[package.PackageIdentity.Id] = versions;
+                    }
+
+                    versions.Add(package.PackageIdentity.Version);
+                }
+            }
+
+            var consolidatablePackages = packages
+                .Where(p => p.Value.Count >= 2)
+                .Select(p => new PackageIdentity(p.Key, p.Value.Max()))
+                .ToList();
+
+            return consolidatablePackages;
+        }
+
+        private async Task<SearchResult> SearchConsolidateAsync(
+            int startIndex, 
+            CancellationToken cancellationToken)
+        {
+            var installedPackages = (await GetConsolidatablePackagesAsync(token: cancellationToken))
+                .Where(p => p.Id.IndexOf(_searchText, StringComparison.OrdinalIgnoreCase) != -1)
+                .OrderBy(p => p.Id)
+                .Skip(startIndex)
+                .Take(_option.PageSize + 1)
+                .ToArray();
+
+            var results = new List<UISearchMetadata>();
+            var localResource = await _packageManager.PackagesFolderSourceRepository
+                .GetResourceAsync<UIMetadataResource>();
+
+            // UIMetadataResource may not be available
+            // Given that this is the 'Installed' filter, we ignore failures in reaching the remote server
+            // Instead, we will use the local UIMetadataResource
+            UIMetadataResource metadataResource;
+            try
+            {
+                metadataResource =
+                _sourceRepository == null ?
+                null :
+                await _sourceRepository.GetResourceAsync<UIMetadataResource>();
+            }
+            catch (Exception ex)
+            {
+                metadataResource = null;
+                // Write stack to activity log
+                Mvs.ActivityLog.LogError(LogEntrySource, ex.ToString());
+            }
+
+            var tasks = new List<Task<UISearchMetadata>>();
+            for (int i = 0; i < installedPackages.Length; i++)
+            {
+                var packageIdentity = installedPackages[i];
+
+                tasks.Add(
+                    Task.Run(() =>
+                        GetPackageMetadataAsync(localResource,
+                                                metadataResource,
+                                                packageIdentity,
+                                                cancellationToken)));
+            }
+
+            await Task.WhenAll(tasks);
```

Comment:
GetPacakgeMetadataAsync is designed for offline use so it never throws when there's server problem.